### PR TITLE
BUGFIX: Extended GPS coordinate import to support just decimal value in .xmp fixes #15444 …

### LINKED ISSUE #15444: Darktable Crashes When Trying to Import An Image From Apple Photos with .xmp File
### Describe the bug

Trying to use import of an image exported from apple photos the GPS coordinates cause darktable to crash.  This is because the code tries to access a list location that doesn't exist. 

I have posted a pull request for code I believe fixes it.  Request number is #15443

### Steps to reproduce

Try to import any image where the associated .xmp files stores GPS locations in this format

```
<exif:GPSLatitudeRef>N</exif:GPSLatitudeRef>
<exif:GPSLatitude>27.942666666666668</exif:GPSLatitude>
<exif:GPSLongitudeRef>W</exif:GPSLongitudeRef>
<exif:GPSLongitude>82.452666666666673</exif:GPSLongitude>
```

### Expected behavior

Properly parse and import the image

### Logfile | Screenshot | Screencast

_No response_

### Commit

It fails in 4.4.2 as well as current master

### Where did you install darktable from?

distro packaging

### darktable version

4.4.2. Also fails on master branch 

### What OS are you using?

Mac

### What is the version of your OS?

MacOS Ventura

### Describe your system?

M2 Pro silicon MacBook Pro
32GB of Ram

### Are you using OpenCL GPU in darktable?

None

### If yes, what is the GPU card and driver?

_No response_

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

_No response_

### PR DESCRIPTION
**Pull request by fgeiger-fnf**
There is an issue where the utility function dt_util_gps_string_to_number assumes that the gps coordinate will be stored in a format where the is a comma, such as DDD,MM.mm{N|S}. 

 However the XMP file that is exported from Apple photos follows the format similar to some cameras where the string relies on two xml keys like:
```
<exif:GPSLatitudeRef>N</exif:GPSLatitudeRef>
<exif:GPSLatitude>27.942666666666668</exif:GPSLatitude>
<exif:GPSLongitudeRef>W</exif:GPSLongitudeRef>
<exif:GPSLongitude>82.452666666666673</exif:GPSLongitude>
```
This causes the code to crash since by default it tries to access list[1] which doesn't exist so the code segfaults.  

These changes extend the system to successfully handle this case.

### CODE REVIEW

**Comment by kmilos** [comment]
Please [link to issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).

**Comment by kmilos** [comment]
> Sorry, it is my first time trying to contribute code (outside of my day job). I added the fixes keyword and issue number in the title but it didn't seem to do anything. I also can't seem to click on the right sidebar on development to try to manually link.

No problem, and thanks for contributing. 👍 

It actually needs to be in your first post/description rather than the title. You can edit it and add "Fixes..." on a line on its own at the top. (Only admins can actually do it through the sidebar directly...)

**Comment by TurboGit** [comment]
@fgeiger-fnf : I have linked the issue.

File: src/develop/lightroom.c [hunk 1]
```diff
@@ -810,12 +811,57 @@ static void _lrop(const dt_develop_t *dev, const xmlDocPtr doc, const dt_imgid_t
         data->has_rating = TRUE;
       }
     }
+    else if(!xmlStrcmp(name, (const xmlChar *)"GPSLatitudeRef"))
+    {
+      if g_str_equal("N", (const char *)value)
```
Comment (TurboGit) [inline src/develop/lightroom.c:816]: This is not C and won't compile. So I bet it has not been tested.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Please fix the syntax errors.

File: src/develop/lightroom.c (hunk 1 above)
Comment (fgeiger-fnf) [reply src/develop/lightroom.c:816]: Let me take a look, I did in fact build and use it.  So let me see what the issue is.

**Comment by fgeiger-fnf** [comment]
OK, definitely weird.  You are correct there should be () after the if, however apple clang doesn't care and works fine.  I have fixed the code so the conditional is inside of the parens as it should be.   I have built it again with apple clang 15 and it still works.  I am in the process of trying to get CLANG 15 from homebrew installed.   I would like to make sure it fails as your linux build does without the ().

**Comment by fgeiger-fnf** [comment]
I have also built with clang 17 from home-brew.  It will build successfully with or without the () so I am now wondering if there are different compiler flag for syntax checking between the home-brew build scripts in packaging/macosx and whatever the CI system is using for the build.

File: src/common/utility.c [hunk 2]
```diff
@@ -568,7 +568,9 @@ double dt_util_gps_string_to_number(const gchar *input)
   gchar **list = g_strsplit(input, ",", 0);
   if(list)
   {
-    if(list[2] == NULL) // format DDD,MM.mm{N|S}
+    if(list[1] == NULL) // Already in decimal format
+     res = g_ascii_strtod(list[0], NULL) ;
```
Comment (TurboGit) [inline src/common/utility.c:572]: Indentation fix needed.

**Review by TurboGit — CHANGES_REQUESTED** [review]
Minor style fix needed. TIA.

File: src/common/utility.c (hunk 2 above)
Comment (fgeiger-fnf) [reply src/common/utility.c]: Sorry, I swear I was trying to make things easier for you all.  Sorry, it has been such a pain.  
Fixed now.

**Comment by TurboGit** [comment]
Please never merge master into a topic branch. For this time and as there is a single commit for the actual fix I'll squash everything together so no need to do anything.

**Review by TurboGit — APPROVED** [review]
Thanks!

**Pull request merged**